samcli/commands/sync/command.py

```diff
@@ -348,14 +348,28 @@ def do_cli(
                     poll_delay=poll_delay,
                     on_failure=None,
                 ) as deploy_context:
-                    with SyncContext(dependency_layer, build_context.build_dir, build_context.cache_dir):
+                    with SyncContext(
+                        dependency_layer, build_context.build_dir, build_context.cache_dir
+                    ) as sync_context:
                         if watch:
                             execute_watch(
-                                template_file, build_context, package_context, deploy_context, dependency_layer, code
+                                template_file,
+                                build_context,
+                                package_context,
+                                deploy_context,
+                                dependency_layer,
+                                code,
+                                sync_context,
```

Comment:
nit: can we combine context objects together (for other places as well) for readability?

Reply:
Updated